packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -27,6 +33,49 @@ class _FuchsiaLogReader extends DeviceLogReader {
   String toString() => name;
 }
 
+class FuchsiaDevices extends PollingDeviceDiscovery {
+  FuchsiaDevices() : super('Fuchsia devices');
+
+  @override
+  bool get supportsPlatform => platform.isLinux || platform.isMacOS;
+
+  @override
+  bool get canListAnything => fuchsiaWorkflow.canListDevices;
+
+  @override
+  Future<List<Device>> pollingGetDevices() async {
+    if (!fuchsiaWorkflow.canListDevices) {
+      return <Device>[];
+    }
+    final String text = await fuchsiaSdk.netls();
+    return parseFuchsiaDeviceOutput(text);
+  }
+
+  @override
+  Future<List<String>> getDiagnostics() async => const <String>[];
+}
+
+/// Parses output from the netls tool into fuchsia devices.
+///
+/// Example output:
+///     $ ./netls
+///     > device liliac-shore-only-last (fe80::82e4:da4d:fe81:227d/3)
+@visibleForTesting
+List<FuchsiaDevice> parseFuchsiaDeviceOutput(String text) {
+  final List<FuchsiaDevice> devices = <FuchsiaDevice>[];
+  for (String rawLine in text.trim().split('\n')) {
+    final String line = rawLine.trim();
+    if (!line.startsWith('device'))
+      continue;
+    // ['device', 'device name', '(id)']
+    final List<String> words = line.split(' ');
+    final String name = words[1];
```

Comment:
Are we sure the name can't contain spaces?  What about the id?  If the name could but the id can't, then can we assume that everything in between the "device" and the id is the name?